criu/ipc_ns.c

```diff
@@ -354,6 +354,57 @@ static int dump_ipc_shm_pages(const IpcShmEntry *shm)
 	return ret;
 }
 
+static int dump_shm_hugetlb_flag(IpcShmEntry *shm, int id)
+{
+	void *addr;
+	int ret, hugetlb_flag, err = -1;
+	char *str;
+	struct bfd f;
+	unsigned int major, minor;
+	unsigned long start;
+
+	addr = shmat(id, NULL, SHM_RDONLY);
+	if (addr == (void *)-1) {
+		pr_perror("Failed to attach shm");
+		return -1;
+	}
+
+	f.fd = open_proc(PROC_SELF, "maps");
```

Comment:
Can we avoid parsing maps for the criu process? We already parse maps for dumped processes. Can we get all required info from there?

Reply:
I try to have a generic approach for dumping device's number of shm as some shm(s) may not be mapped (the maps of dumped process does not have device's number of these shm(s)). I'm looking at shmctl but can't not see any ways to get device's number. So my solution is when collecting shm identifier, we need to map that shm then get the device's number.
Your concern is right, I didn't think about it. I try to change to use stat on map_files symlink instead
https://github.com/minhbq-99/criu/commit/7f252688754d1c943b6ba0142ba8fb99595c120e

But I get CI error in userfaultfd which I still don't understand.